.github/scripts/nft_boost/activateBoost.ts

```diff
@@ -0,0 +1,226 @@
+          address: gauge,
+          abi: GaugeAbi,
+          functionName: 'estimatedBackerRewards',
+          args: [token, backer],
+        })),
+      })
+      gaugeEstimatedRewards = gaugeEstimatedRewards.map(data => data.result)
+      return {
+        RBTC: BigInt(gaugeEstimatedRewards[0]),
+        RIF: BigInt(gaugeEstimatedRewards[1]),
+      }
+    } catch (error) {
+      console.error(`Error fetching rewards for ${backer}:`, error)
+      throw error
+    }
+  }
+
+  return {
+    getLatestBlockNumber,
+    getNftTransferEvents,
+    getAllGauges,
+    estimatedGaugeRewards,
+  }
+}
+
+async function main() {
+  const { nftContractAddress, boostPercentage, env } = args.reduce<Args>((acc, val) => {
+    if (val.startsWith('--nft')) {
+      acc.nftContractAddress = getAddress(val.split('=')[1])
+    }
+    if (val.startsWith('--boost')) {
+      acc.boostPercentage = parseFloat(val.split('=')[1])
+    }
+    if (val.startsWith('--env')) {
+      acc.env = val.split('=')[1]
+    }
+    return acc
+  }, {} as Args)
+
+  if (!env || !nftContractAddress || isNaN(boostPercentage)) {
+    throw new Error(
+      'Usage: npx tsx .github/scripts/nft_boost/activateBoost.ts \\\
+      --nft=<nftContractAddress> \\\
+      --boost=<boostPercentage> \\\
+      --env=<env>',
+    )
+  }
+  console.info('nft: ', nftContractAddress)
+  console.info('Boost percentage: ', boostPercentage)
+  console.info('env: ', env)
+  if (boostPercentage < 0) {
+    throw new Error('Boost percentage must be positive')
+  }
+
+  envConfig({
+    path: `.env.${env}`,
+  })
+
+  const { getLatestBlockNumber, getNftTransferEvents, getAllGauges, estimatedGaugeRewards } =
+    await getActions()
+
+  const blockNumber = await getLatestBlockNumber()
+  console.info(`Latest block: ${blockNumber}`)
+
+  const nftTransferEvents = await getNftTransferEvents(nftContractAddress)
+  const holdersData: any = {}
+
+  const gauges = await getAllGauges()
+
+  const normalisationFactor = 100_000 // Scaling factor to handle decimal values in BigInts. A ratio of 1.25 => 125_000
+  const boost = BigInt(Math.round(normalisationFactor * (1 + boostPercentage / 100)))
+
+  for (let i = 0; i < nftTransferEvents.length; i++) {
+    const event = nftTransferEvents[i]
+    const holderAddress = getAddress(`0x${event.topics[2].slice(-40)}`)
+    console.info(`Processing ${i + 1} of ${nftTransferEvents.length} events. Nft holder: ${holderAddress}`)
+    const tokenId = BigInt(event.topics[3]).toString()
+
+    let backerEstimatedRBTCRewards = 0n
+    let backerEstimatedRIFRewards = 0n
+    let backerBoostedRBTCRewards = 0n
+    let backerBoostedRIFRewards = 0n
+    for (const gauge of gauges) {
+      const estimatedRewards = await estimatedGaugeRewards(holderAddress, gauge)
+
+      const boostedRBTCRewards = (estimatedRewards.RBTC * boost) / BigInt(normalisationFactor)
+      const boostedRIFRewards = (estimatedRewards.RIF * boost) / BigInt(normalisationFactor)
```

Comment:
doesn't make more sense to calculated the boosted reward after the lop rather than for each gauge?